packages/react-reconciler/src/ReactFiberNewContext.new.js

```diff
@@ -157,12 +158,34 @@ export function scheduleWorkOnParentPath(
     ) {
       alternate.childLanes = mergeLanes(alternate.childLanes, renderLanes);
     } else {
-      // Neither alternate was updated, which means the rest of the
+      // Neither alternate was updated.
+      // Normally, this would mean that the rest of the
       // ancestor path already has sufficient priority.
+      // However, this is not necessarily true inside offscreen
+      // or fallback trees because childLanes may be inconsistent
+      // with the surroundings. This is why we continue the loop.
+    }
+    if (node === propagationRoot) {
       break;
     }
+    if (__DEV__) {
+      if (node === propagationRoot.alternate) {
```

Comment:
Nit: The warning below seems sufficient now? It would only catch stuff at the propagation root anyway.